### PR TITLE
#8751: move loginController inside the useEffect

### PR DESCRIPTION
**Pull request by grahamlangford**
## What does this PR do?

- Follows https://github.com/pixiebrix/pixiebrix-extension/pull/8803/files/122e7a1cff02bd51151d47ba6572c1e2f895f7db..8962a07991b435349e97f80c254cefeeca104a5b#r1674693657

For more information on our expectations for the PR process, see the
[code review principles doc](https://www.notion.so/pixiebrix/Code-Review-Principles-1ce7276b82a84d2a995d55ad85e1310d?pvs=4)

### CODE REVIEW

File: src/contrib/google/sheets/core/useGoogleAccount.ts [hunk 1]
```diff
@@ -28,13 +28,14 @@ import { oauth2Storage } from "@/auth/authConstants";
...
 
 const GOOGLE_PKCE_INTEGRATION_ID = validateRegistryId("google/oauth2-pkce");
-const loginController = new ReusableAbortController();
 
 function useGoogleAccountLoginListener({
   data: googleAccount,
   refetch,
 }: FetchableAsyncState<SanitizedIntegrationConfig | null>) {
   useEffect(() => {
+    const loginController = new ReusableAbortController();
```
Comment (twschiller) [inline src/contrib/google/sheets/core/useGoogleAccount.ts:37]: Can just use a normal abort controller. Doesn't need to be reusable

Comment (grahamlangford) [reply src/contrib/google/sheets/core/useGoogleAccount.ts:37]: I think you're right. I need to send the signal as the second arg for cleanup, but I don't need to abort the handler until unmount

**Review by twschiller — APPROVED** [review]
LGTM

For reference, here's our `useEventListener` hook for custom events/dom events. I don't think it's possible to use for `oauth2Storage.onChanged though`

**Comment by grahamlangford** [comment]
> oauth2Storage

Good to know. Yeah, the onChanged is already a lot of customization to improve the ergonomics. Useful to know about for other events though

**Pull request merged**